skglm/solvers/cd_solver.py

```diff
@@ -235,38 +228,43 @@ def cd_solver(
     stop_crit = np.inf  # initialize for case n_iter=0
 
     is_sparse = sparse.issparse(X)
+
+    if len(w) != n_features + fit_intercept:
+        raise ValueError(
+            "Shape of w should be n_features + 1 when fit_intercept=True.")
```

Comment:
nice
Can you add `" expected %d, got %d" ` to make the message crystal clear for the user?